inbm/cloudadapter-agent/cloudadapter/cloud/adapters/inbs_adapter.py

```diff
@@ -0,0 +1,62 @@
+"""
+Adapter for communication with the cloud agent on the device. It abstracts
+creation of the cloud connection, termination, creating commands etc.
+
+Connects to INBS service via gRPC
+
+Copyright (C) 2017-2024 Intel Corporation
+SPDX-License-Identifier: Apache-2.0
+"""
+
+from ...exceptions import AdapterConfigureError
+from ...cloud.client.inbs_cloud_client import InbsCloudClient
+from ..client.cloud_client import CloudClient
+from .adapter import Adapter
+from typing import Callable
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class InbsAdapter(Adapter):
+    def __init__(self, configs: dict) -> None:
+        super().__init__(configs)
+
+    def configure(self, configs: dict) -> CloudClient:
+        """Configure the INBS cloud adapter
+
+        @param configs: schema conforming JSON config data
+        @exception AdapterConfigureError: If configuration fails
+        """
+        hostname = configs.get("hostname")
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

For scalability and to avoid hardcoding, consider externalizing `hostname`, `port`, `node-id`, and `token` into a configuration file or environment variables.

Reply:
configs is already coming from a configuration file